### PR TITLE
adding tests (with refacto)

### PR DESCRIPTION
**Pull request by prhod**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+                r_id.push_str("Line:Relation:");
+            } else {
+                r_id.push_str("Route:Relation:");
+            }
+            r_id.push_str(&rel.id.to_string());
+            Box::new(Route{id: r_id,
+                      name: rel.tags.get("name").unwrap_or(&"".to_string()).to_string(),
+                      code: rel.tags.get("ref").unwrap_or(&"".to_string()).to_string() })
+        },
+        _ => Box::new({ Route{id: "error".to_string(),
```
Comment (kinnou02) [inline src/lib.rs]: You should return a `Result` or an `Option`

File: src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+        osmpbfreader::OsmObj::Relation(ref rel)=> {
+            let mut r_id : String = "".to_string();
+            if rel.tags.get("type").unwrap_or(&"".to_string()) == "route_master" {
+                r_id.push_str("Line:Relation:");
+            } else {
+                r_id.push_str("Route:Relation:");
+            }
+            r_id.push_str(&rel.id.to_string());
+            Box::new(Route{id: r_id,
+                      name: rel.tags.get("name").unwrap_or(&"".to_string()).to_string(),
```
Comment (kinnou02) [inline src/lib.rs]: It should be an `Option`

File: src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+        }
+        osmpbfreader::OsmObj::Node(ref node) => {
+            let mut sp_id : String = "StopPoint:Node:".to_string();
+            sp_id.push_str(&node.id.to_string());
+            Box::new(StopPoint{id: sp_id.to_string(),
+                      name: node.tags.get("name").unwrap_or(&"".to_string()).to_string(),
+                      coord: Coord{lat: node.lat, lon: node.lon} } )
+        }
+    }
+}
```
Comment (TeXitoi) [inline src/lib.rs]:
```rust
fn osm_obj_2_stop_point(obj_map: &BTreeMap<osmpbfreader::OsmId, osmpbfreader::OsmObj>,
                        obj: &osmpbfreader::OsmObj)
                        -> StopPoint {
    use osmpbfreader::OsmObj::*;
    let (kind, id, coord) = match *obj {
        Relation(ref rel) => ("Relation", rel.id, get_rel_coord(obj_map, rel)),
        Way(ref way) => ("Way", way.id, get_way_coord(obj_map, way)),
        Node(ref node) => ("Node", node.id, Coord{ lat: node.lat, lon: node.lon }),
    };
    let id = format!("StopPoint:{}:{}", kind, id);
    let name = obj.tags().get("name").cloned().unwrap_or("".to_string());
    StopPoint { id: id, name: name, coord: coord }
}
```

File: src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+    let objects = osmpbfreader::get_objs_and_deps(&mut parsed_pbf, is_stop_point).unwrap();
+
+    for (_, obj) in &objects {
+        if !is_stop_point(&obj) {
+            continue;
+        }
+        let sp = osm_obj_2_stop_point(&objects, obj);
+        stops_vect.push(sp);
+    }
+    stops_vect
```
Comment (TeXitoi) [inline src/lib.rs]:
```rust
pub fn get_stops_from_osm(mut parsed_pbf: &mut OsmPbfReader) -> Vec<StopPoint> {
    let objects = osmpbfreader::get_objs_and_deps(&mut parsed_pbf, is_stop_point).unwrap();
    objects.iter()
        .filter(|(_, obj)| is_stop_point(obj))
        .map(|(_, obj) osm_obj_2_stop_point(&objects, obj))
        .collect()
}
````

Same pattern for `get_routes_from_osm()`

File: src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+// License along with this program. If not, see
+// <http://www.gnu.org/licenses/>.
+//
+// Stay tuned using
+// twitter @navitia
+// IRC #navitia on freenode
+// https://groups.google.com/d/forum/navitia
+// www.navitia.io
+
+#[macro_use]
```
Comment (TeXitoi) [inline src/lib.rs]: no macros in osmpbfreader, just remove this line

File: src/lib.rs [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+    routes_vect
+}
+
+pub fn write_stops_to_csv(stops : Vec<Box<StopPoint>>) {
+    let csv_file = std::path::Path::new("/tmp/osmtc2mongo.csv");
+    let mut wtr = csv::Writer::from_file(csv_file).unwrap();
+
+    for sp in &stops {
+        let result = wtr.encode(sp);
+        assert!(result.is_ok());
```
Comment (TeXitoi) [inline src/lib.rs]: better to unwrap the result, the error will be better than an assert:
```rust
wtr.encode(sp).unwrap();
```

File: src/lib.rs [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+            rel.tags.get("type").map_or(false, |v| v == "route") ||
+            rel.tags.get("type").map_or(false, |v| v == "route_master")
+        },
+        _ => false,
+    }
+}
+
+fn get_way_coord(obj_map: &BTreeMap<osmpbfreader::OsmId, osmpbfreader::OsmObj>,
+                 way: &osmpbfreader::objects::Way)
+                 -> Coord {
```
Comment (prhod) [inline src/lib.rs]: no, the purpose of this function is to provide a unique coordinate to a way stop_point. Do you want me to rename the func ?

Comment (kinnou02) [inline src/lib.rs]: I was expecting it to be used for route, at least a comment would be great I think

File: src/lib.rs [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+fn get_one_coord_from_rel(obj_map: &BTreeMap<osmpbfreader::OsmId, osmpbfreader::OsmObj>,
+                 rel: &osmpbfreader::objects::Relation)
+                 -> Coord {
+    rel.refs
+        .iter()
+        .filter_map(|refe| {
+            obj_map.get(&refe.member).or_else(|| {
+                None
+            })
+        })
```
Comment (TeXitoi) [inline src/lib.rs]:
```rust
.filter_map(|refe| obj_map.get(&refe.member))
```

File: src/lib.rs [hunk 9]
```diff
@@ -0,0 +1,179 @@
...
+        Relation(ref rel)=> {
+            let kind = if rel.tags.get("type").map_or(false, |v| v == "route_master") {
+                "Line"
+            } else {
+                "Route"
+            };
+            let r_id = format!("{}:Relation:{}", kind, rel.id);
+            Some(Route{id: r_id,
+                      name: rel.tags.get("name").unwrap_or(&"".to_string()).to_string(),
+                      code: rel.tags.get("ref").unwrap_or(&"".to_string()).to_string() })
```
Comment (TeXitoi) [inline src/lib.rs]: as before for this pattern:
```rust
rel.tags.get("ref").cloned().unwrap_or("".to_string())
```

File: src/lib.rs [hunk 10]
```diff
@@ -0,0 +1,179 @@
...
+}
+
+
+fn osm_obj_to_stop_point(obj_map: &BTreeMap<osmpbfreader::OsmId, osmpbfreader::OsmObj>,
+                        obj: &osmpbfreader::OsmObj)
+                        -> StopPoint {
+    let (obj_type, obj_id, coord) = match *obj {
+        Relation(ref rel) => ("Relation", rel.id, get_one_coord_from_rel(obj_map, rel)),
+        Way(ref way) => ("Way", way.id, get_one_coord_from_way(obj_map, way)),
+        Node(ref node) => ("Node", node.id, Coord{lat: node.lat, lon: node.lon})
```
Comment (TeXitoi) [inline src/lib.rs]: Canonical spacing is (not repeated for the other occurences)
```rust
, Coord { lat: node.lat, lon: node.lon })
```

**Pull request merged**